src/sdk/pynni/nni/pbt_tuner/pbt_tuner.py

```diff
@@ -34,19 +38,102 @@ def exploit_and_explore(bot_trial_info, top_trial_info, factors, epoch, search_s
     bot_checkpoint_dir = bot_trial_info.checkpoint_dir
     top_hyper_parameters = top_trial_info.hyper_parameters
     hyper_parameters = copy.deepcopy(top_hyper_parameters)
-    # TODO think about different type of hyperparameters for 1.perturbation 2.within search space
+    random_state = np.random.RandomState()
     for key in hyper_parameters.keys():
         if key == 'load_checkpoint_dir':
             hyper_parameters[key] = hyper_parameters['save_checkpoint_dir']
         elif key == 'save_checkpoint_dir':
             hyper_parameters[key] = os.path.join(bot_checkpoint_dir, str(epoch))
-        elif isinstance(hyper_parameters[key], float):
-            perturb = np.random.choice(factors)
-            val = hyper_parameters[key] * perturb
+        elif search_space[key]["_type"] == "choice":
+            choices = search_space[key]["_value"]
+            can_sort = True
+            for item in choices:
+                if isinstance(item, str):
+                    can_sort = False
+            if can_sort:
+                choices.sort()
```

Comment:
can be replaced with the following code:

```python
try:
    choices.sort()
except Exception as e:
    logger.warning("Choice cannot be sorted, %s", e.message)
```

Reply:
modified